### PR TITLE
Start of extracting interfaces.

### PR DESCRIPTION
**Pull request by RichardLake**
See #1149

### CODE REVIEW

**Comment by pjf** [comment]
I've been in flux the last 24 hours, but the first 30% of this I've looked through seems excellent. Thank you so much!

File: Core/Registry/IRegistryQuerier.cs [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+        /// </summary>
+        /// <param name="with_provides">If set to false will not check for provided versions.</param>
+        /// <returns>The version of the mod or null if not found</returns>
+        Version InstalledVersion(string identifier, bool with_provides = true);
+    }
+
+    /// <summary>
+    /// Helpers for <see cref="IRegistryQuerier"/>
+    /// </summary>
+    public static class IRegistryQuerierHelpers
```
Comment (pjf) [inline Core/Registry/IRegistryQuerier.cs]: I'm new to the culture, but is `IRegistryQuerierHelpers` the accepted way of doing this? The `I` immediately makes me think this is a virtual (interface) class, except that obviously that's not the case here.

(In Moose we'd compile the virtual class and the helper into a *role*, but obviously we can't do that in C#. :)

**Comment by pjf** [comment]
This is great! Merged with thanks!

File: Core/Registry/IRegistryQuerier.cs (hunk 1 above)
Comment (RichardLake) [inline Core/Registry/IRegistryQuerier.cs]: You are correct on the naming of the helper class - I'm just used to
postfixing helper and forgot about the I prefix. Given that it only
contains extension methods a postfix of extensions may be better then
helper anyway. I'm currently interstate so I can't make a PR untill Sunday.

**Pull request merged**